### PR TITLE
Add cuda implementation for `prelu` backward pass

### LINKED ISSUE #18601: prelu这个op没有gpu反向实现
prelu这个op只实现了gpu前向，没有gpu反向，导致训练迭代速度降为原来的五分之一到六分之一。

### PR DESCRIPTION
**Pull request by willthefrog**
fix #18601

### CODE REVIEW

**Comment by jerrywgz** [comment]
Need add test=develop in commit comments.

File: paddle/fluid/operators/prelu_op.cu [hunk 1]
```diff
@@ -55,10 +59,116 @@ class CUDAPReluKernel : public framework::OpKernel<T> {
...
+    const T* alpha_ptr = alpha->data<T>();
+    const T* dy_ptr = dy->data<T>();
+    T* dx_ptr = dx ? dx->mutable_data<T>(context.GetPlace()) : nullptr;
+    T* dalpha_ptr =
+        dalpha ? dalpha->mutable_data<T>(context.GetPlace()) : nullptr;
+
+    auto& mode = context.Attr<std::string>("mode");
+
+    int numel = x->numel();
+    auto dim = x->dims();
```
Comment (jerrywgz) [inline paddle/fluid/operators/prelu_op.cu:159]: Return type is DDim, don't need convert to vector

File: paddle/fluid/operators/prelu_op.cu [hunk 2]
```diff
@@ -55,10 +59,116 @@ class CUDAPReluKernel : public framework::OpKernel<T> {
...
+    } else if (dalpha_ptr == nullptr) {
+      dalpha_tmp_ptr = nullptr;
+    } else {
+      auto& dev_ctx = context.template device_context<DeviceContext>();
+      dalpha_tmp =
+          context.AllocateTmpTensor<T, DeviceContext>(input_dim_, dev_ctx);
+      dalpha_tmp_ptr = dalpha_tmp.mutable_data<T>(context.GetPlace());
+    }
+
+    if (mode == "element" || dalpha_ptr == nullptr) {
```
Comment (jerrywgz) [inline paddle/fluid/operators/prelu_op.cu]: Need check cases for mode="channel" and "all".

File: paddle/fluid/operators/prelu_op.cu (hunk 1 above)
Comment (willthefrog) [reply paddle/fluid/operators/prelu_op.cu:159]: done

File: paddle/fluid/operators/prelu_op.cu (hunk 2 above)
Comment (willthefrog) [reply paddle/fluid/operators/prelu_op.cu]: done

File: paddle/fluid/operators/prelu_op.cu [hunk 3]
```diff
@@ -55,10 +59,112 @@ class CUDAPReluKernel : public framework::OpKernel<T> {
...
+      auto& dev_ctx = context.template device_context<DeviceContext>();
+      dalpha_tmp = context.AllocateTmpTensor<T, DeviceContext>(dim, dev_ctx);
+      dalpha_tmp_ptr = dalpha_tmp.mutable_data<T>(context.GetPlace());
+    }
+
+    PreluGradElementwiseFunctor<T> prelu_element_wise_grad;
+    prelu_element_wise_grad(stream, x_ptr, y_ptr, alpha_ptr, dy_ptr, dx_ptr,
+                            dalpha_tmp_ptr, input_shape);
+
+    if (mode == "element" || dalpha_ptr == nullptr) return;
```
Comment (jerrywgz) [inline paddle/fluid/operators/prelu_op.cu]: 这里如果dalpha_ptr=nullptr mode=“channel”，dx_ptr的计算方法还是有问题吧

Comment (willthefrog) [reply paddle/fluid/operators/prelu_op.cu]: fixed

**Review by jerrywgz — APPROVED** [review]
LGTM

**Pull request merged**